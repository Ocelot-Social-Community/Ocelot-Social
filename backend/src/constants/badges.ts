// this file is duplicated in `backend/src/constants/badges` and `webapp/constants/badges.js`
export const TROPHY_BADGES_SELECTED_MAX = 9
