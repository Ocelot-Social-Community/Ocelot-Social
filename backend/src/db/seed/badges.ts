import Factory from '@db/factories'

export const trophies = async () => {
  return {
    // Blue Animals
    trophyBear: await Factory.build('badge', {
      id: 'trophy_bear',
      type: 'trophy',
      description: 'You earned a Bear',
      icon: '/img/badges/trophy_blue_bear.svg',
    }),
    trophyPanda: await Factory.build('badge', {
      id: 'trophy_panda',
      type: 'trophy',
      description: 'You earned a Panda',
      icon: '/img/badges/trophy_blue_panda.svg',
    }),
    trophyRabbit: await Factory.build('badge', {
      id: 'trophy_rabbit',
      type: 'trophy',
      description: 'You earned a Rabbit',
      icon: '/img/badges/trophy_blue_rabbit.svg',
    }),
    trophyRacoon: await Factory.build('badge', {
      id: 'trophy_racoon',
      type: 'trophy',
      description: 'You earned a Racoon',
      icon: '/img/badges/trophy_blue_racoon.svg',
    }),
    trophyRhino: await Factory.build('badge', {
      id: 'trophy_rhino',
      type: 'trophy',
      description: 'You earned a Rhino',
      icon: '/img/badges/trophy_blue_rhino.svg',
    }),
    trophyTiger: await Factory.build('badge', {
      id: 'trophy_tiger',
      type: 'trophy',
      description: 'You earned a Tiger',
      icon: '/img/badges/trophy_blue_tiger.svg',
    }),
    trophyTurtle: await Factory.build('badge', {
      id: 'trophy_turtle',
      type: 'trophy',
      description: 'You earned a Turtle',
      icon: '/img/badges/trophy_blue_turtle.svg',
    }),
    trophyWhale: await Factory.build('badge', {
      id: 'trophy_whale',
      type: 'trophy',
      description: 'You earned a Whale',
      icon: '/img/badges/trophy_blue_whale.svg',
    }),
    trophyWolf: await Factory.build('badge', {
      id: 'trophy_wolf',
      type: 'trophy',
      description: 'You earned a Wolf',
      icon: '/img/badges/trophy_blue_wolf.svg',
    }),
    // Green Transports
    trophyAirship: await Factory.build('badge', {
      id: 'trophy_airship',
      type: 'trophy',
      description: 'You earned an Airship',
      icon: '/img/badges/trophy_green_airship.svg',
    }),
    trophyAlienship: await Factory.build('badge', {
      id: 'trophy_alienship',
      type: 'trophy',
      description: 'You earned an Alienship',
      icon: '/img/badges/trophy_green_alienship.svg',
    }),
    trophyBalloon: await Factory.build('badge', {
      id: 'trophy_balloon',
      type: 'trophy',
      description: 'You earned a Balloon',
      icon: '/img/badges/trophy_green_balloon.svg',
    }),
    trophyBigballoon: await Factory.build('badge', {
      id: 'trophy_bigballoon',
      type: 'trophy',
      description: 'You earned a Big Balloon',
      icon: '/img/badges/trophy_green_bigballoon.svg',
    }),
    trophyCrane: await Factory.build('badge', {
      id: 'trophy_crane',
      type: 'trophy',
      description: 'You earned a Crane',
      icon: '/img/badges/trophy_green_crane.svg',
    }),
    trophyGlider: await Factory.build('badge', {
      id: 'trophy_glider',
      type: 'trophy',
      description: 'You earned a Glider',
      icon: '/img/badges/trophy_green_glider.svg',
    }),
    trophyHelicopter: await Factory.build('badge', {
      id: 'trophy_helicopter',
      type: 'trophy',
      description: 'You earned a Helicopter',
      icon: '/img/badges/trophy_green_helicopter.svg',
    }),
    // Green Animals
    trophyBee: await Factory.build('badge', {
      id: 'trophy_bee',
      type: 'trophy',
      description: 'You earned a Bee',
      icon: '/img/badges/trophy_green_bee.svg',
    }),
    trophyButterfly: await Factory.build('badge', {
      id: 'trophy_butterfly',
      type: 'trophy',
      description: 'You earned a Butterfly',
      icon: '/img/badges/trophy_green_butterfly.svg',
    }),
    // Green Plants
    trophyFlower: await Factory.build('badge', {
      id: 'trophy_flower',
      type: 'trophy',
      description: 'You earned a Flower',
      icon: '/img/badges/trophy_green_flower.svg',
    }),
    trophyLifetree: await Factory.build('badge', {
      id: 'trophy_lifetree',
      type: 'trophy',
      description: 'You earned the tree of life',
      icon: '/img/badges/trophy_green_lifetree.svg',
    }),
    // Green Misc
    trophyDoublerainbow: await Factory.build('badge', {
      id: 'trophy_doublerainbow',
      type: 'trophy',
      description: 'You earned the Double Rainbow',
      icon: '/img/badges/trophy_green_doublerainbow.svg',
    }),
    trophyEndrainbow: await Factory.build('badge', {
      id: 'trophy_endrainbow',
      type: 'trophy',
      description: 'You earned the End of the Rainbow',
      icon: '/img/badges/trophy_green_endrainbow.svg',
    }),
    trophyMagicrainbow: await Factory.build('badge', {
      id: 'trophy_magicrainbow',
      type: 'trophy',
      description: 'You earned the Magic Rainbow',
      icon: '/img/badges/trophy_green_magicrainbow.svg',
    }),
    trophyStarter: await Factory.build('badge', {
      id: 'trophy_starter',
      type: 'trophy',
      description: 'You earned the Starter Badge',
      icon: '/img/badges/trophy_green_starter.svg',
    }),
    trophySuperfounder: await Factory.build('badge', {
      id: 'trophy_superfounder',
      type: 'trophy',
      description: 'You earned the Super Founder Badge',
      icon: '/img/badges/trophy_green_superfounder.svg',
    }),
  }
}

export const verification = async () => {
  return {
    // Red Role
    verificationModerator: await Factory.build('badge', {
      id: 'verification_moderator',
      type: 'verification',
      description: 'You are a Moderator',
      icon: '/img/badges/verification_red_moderator.svg',
    }),
    verificationAdmin: await Factory.build('badge', {
      id: 'verification_admin',
      type: 'verification',
      description: 'You are an Administrator',
      icon: '/img/badges/verification_red_admin.svg',
    }),
    verificationDeveloper: await Factory.build('badge', {
      id: 'verification_developer',
      type: 'verification',
      description: 'You are a Developer',
      icon: '/img/badges/verification_red_developer.svg',
    }),
  }
}
