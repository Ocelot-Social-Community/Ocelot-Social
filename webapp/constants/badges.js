// this file is duplicated in `backend/src/constants/badges` and `webapp/constants/badges.js`
export const PROFILE_BADGE_COUNT = 9
