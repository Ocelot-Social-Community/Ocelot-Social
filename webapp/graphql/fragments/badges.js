import gql from 'graphql-tag'

export const badges = gql`
  fragment badges on User {
    badgeTrophiesSelected {
      id
      icon
      description
    }
    badgeVerification {
      id
      icon
      description
    }
  }
`
