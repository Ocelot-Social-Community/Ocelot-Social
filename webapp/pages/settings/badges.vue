<template>
  <base-card>
    <h2 class="title">{{ $t('settings.badges.name') }}</h2>
    <ds-space margin-bottom="big">
      <div class="presenterContainer">
        <badges
          :badges="[currentUser.badgeVerification, ...selectedBadges]"
          :selection-mode="true"
          :scale="2"
          @badge-selected="handleBadgeSlotSelection"
          ref="badgesComponent"
        />
      </div>

      <div v-if="selectedBadgeIndex !== null && !isEmptySlot" class="badge-actions">
        <base-button @click="removeBadgeFromSlot" class="remove-button">
          {{ $t('settings.badges.remove') }}
        </base-button>
      </div>

      <div v-if="selectedBadgeIndex !== null && isEmptySlot" class="selection-info">
        <badge-selection
          :badges="availableBadges"
          @badge-selected="assignBadgeToSlot"
          ref="badgeSelection"
        />
      </div>
    </ds-space>
  </base-card>
</template>

<script>
import { mapGetters, mapMutations } from 'vuex'
import { setTrophyBadgeSelected } from '~/graphql/User'
import scrollToContent from './scroll-to-content.js'
import Badges from '../../components/Badges.vue'
import BadgeSelection from '../../components/BadgeSelection.vue'

export default {
  components: { BadgeSelection, Badges },
  mixins: [scrollToContent],
  data() {
    return {
      selectedBadgeIndex: null,
    }
  },
  computed: {
    ...mapGetters({
      currentUser: 'auth/user',
    }),
    selectedBadges() {
      return this.currentUser.badgeTrophiesSelected
    },
    availableBadges() {
      return this.currentUser.badgeTrophiesUnused
    },
    isEmptySlot() {
      console.log('selectedBadgeIndex', this.selectedBadgeIndex)
      console.log('selectedBadges', this.selectedBadges)
      return this.selectedBadges[this.selectedBadgeIndex]?.isDefault ?? true
    },
  },
  created() {
    this.userBadges = [...(this.currentUser.badgeTrophiesSelected || [])]
  },
  methods: {
    ...mapMutations({
      setCurrentUser: 'auth/SET_USER',
    }),
    handleBadgeSlotSelection(index) {
      if (index === 0) {
        this.$toast.info(
          this.$t('settings.badges.verification') ||
            'Dies ist deine Verifikations-Badge und kann nicht geÃ¤ndert werden.',
        )
        this.$refs.badgesComponent.resetSelection()
        return
      }

      this.selectedBadgeIndex = index === null ? null : index - 1 // The first badge in badges component is the verification badge
    },
    async setSlot(badge, slot) {
      await this.$apollo.mutate({
        mutation: setTrophyBadgeSelected,
        variables: {
          badgeId: badge?.id ?? null,
          slot,
        },
        update: (_, { data: { setTrophyBadgeSelected } }) => {
          const { badgeTrophiesSelected, badgeTrophiesUnused } = setTrophyBadgeSelected
          this.setCurrentUser({
            ...this.currentUser,
            badgeTrophiesSelected,
            badgeTrophiesUnused,
          })
        },
      })
    },
    async assignBadgeToSlot(badge) {
      if (!badge || this.selectedBadgeIndex === null) {
        return
      }

      await this.setSlot(badge, this.selectedBadgeIndex)

      this.$toast.success(this.$t('settings.badges.success-update'))

      if (this.$refs.badgeSelection && this.$refs.badgeSelection.resetSelection) {
        this.$refs.badgeSelection.resetSelection()
      }
      this.$refs.badgesComponent.resetSelection()
      this.selectedBadgeIndex = null
    },
    async removeBadgeFromSlot() {
      if (this.selectedBadgeIndex === null) return

      await this.setSlot(null, this.selectedBadgeIndex)

      this.$refs.badgesComponent.resetSelection()
      this.selectedBadgeIndex = null
    },
  },
}
</script>

<style lang="scss">
.presenterContainer {
  padding-top: 50px;
  min-height: 250px;
}

.badge-actions {
  margin-top: 20px;
  display: flex;
  justify-content: center;

  .remove-button {
    margin-top: 8px;
  }
}

.selection-info {
  margin-top: 20px;
  padding: 16px;
}
</style>
