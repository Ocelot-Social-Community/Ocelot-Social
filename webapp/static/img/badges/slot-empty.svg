<?xml version="1.0" encoding="UTF-8"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->
<svg width="400" height="346.67" version="1.1" viewBox="0 0 400 346.67" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
 <defs>
  <linearGradient id="linearGradient4" x1="708.76" x2="493.17" y1="280.91" y2="65.326" gradientTransform="translate(-404.06 .215)" gradientUnits="userSpaceOnUse">
   <stop stop-color="#c1c1c1" offset="0"/>
   <stop stop-color="#fcfcfc" offset="1"/>
  </linearGradient>
 </defs>
 <path d="m-0.21505 173.98 100.65-173.76h198.71l101.08 173.76-99.785 172.04-201.29 0.43011z" fill="#bebebe"/>
 <path d="m22.482 173.91 89.236-154.07h176.18l89.617 154.07-88.473 152.54-178.47 0.38135z" fill="url(#linearGradient4)"/>
</svg>
