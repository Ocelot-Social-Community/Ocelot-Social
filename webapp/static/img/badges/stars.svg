<?xml version="1.0" encoding="UTF-8"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->
<svg width="170mm" height="65mm" version="1.1" viewBox="0 0 170 65" xmlns="http://www.w3.org/2000/svg">
 <g transform="translate(-40.481 -87.048)" fill="#f2bd27" stroke-width="1.5118">
  <path transform="matrix(.30472 .0030284 -.0030284 .30472 .96713 112.56)" d="m461.32 72.567-52.541-26.955-51.991 28.001 9.3998-58.299-42.696-40.794 58.35-9.0756 25.603-53.213 26.663 52.69 58.52 7.9067-41.872 41.64z"/>
  <path transform="matrix(.21871 .0021736 -.0021736 .21871 75.918 124.75)" d="m461.32 72.567-52.541-26.955-51.991 28.001 9.3998-58.299-42.696-40.794 58.35-9.0756 25.603-53.213 26.663 52.69 58.52 7.9067-41.872 41.64z"/>
  <path transform="matrix(.21871 .0021736 -.0021736 .21871 -3.5617 124.75)" d="m461.32 72.567-52.541-26.955-51.991 28.001 9.3998-58.299-42.696-40.794 58.35-9.0756 25.603-53.213 26.663 52.69 58.52 7.9067-41.872 41.64z"/>
  <path transform="matrix(.16908 .0016804 -.0016804 .16908 -13.934 138.16)" d="m461.32 72.567-52.541-26.955-51.991 28.001 9.3998-58.299-42.696-40.794 58.35-9.0756 25.603-53.213 26.663 52.69 58.52 7.9067-41.872 41.64z"/>
  <path transform="matrix(.16908 .0016804 -.0016804 .16908 126.82 138.16)" d="m461.32 72.567-52.541-26.955-51.991 28.001 9.3998-58.299-42.696-40.794 58.35-9.0756 25.603-53.213 26.663 52.69 58.52 7.9067-41.872 41.64z"/>
 </g>
</svg>
