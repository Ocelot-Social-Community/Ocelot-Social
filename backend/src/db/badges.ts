import { getNeode } from './neo4j'
import { trophies, verification } from './seed/badges'

// eslint-disable-next-line import/newline-after-import
;(async function () {
  const neode = getNeode()
  try {
    await trophies()
    await verification()
  } finally {
    await neode.close()
  }
})()
