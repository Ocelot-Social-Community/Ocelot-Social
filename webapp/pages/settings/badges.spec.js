import { render, screen, fireEvent } from '@testing-library/vue'
import '@testing-library/jest-dom'
import badges from './badges.vue'

const localVue = global.localVue

describe('badge settings', () => {
  let mocks

  const apolloMutateMock = jest.fn()

  const Wrapper = () => {
    return render(badges, {
      localVue,
      mocks,
    })
  }

  beforeEach(() => {
    mocks = {
      $t: jest.fn((t) => t),
      $toast: {
        success: jest.fn(),
        error: jest.fn(),
      },
      $apollo: {
        mutate: apolloMutateMock,
      },
    }
  })

  describe('without badges', () => {
    beforeEach(() => {
      mocks.$store = {
        getters: {
          'auth/isModerator': () => false,
          'auth/user': {
            id: 'u23',
            badgeVerification: {
              id: 'bv1',
              icon: '/verification/icon',
              description: 'Verification description',
              isDefault: true,
            },
            badgeTrophiesSelected: [],
            badgeTrophiesUnused: [],
          },
        },
      }
    })

    it('renders', () => {
      const wrapper = Wrapper()
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  describe('with badges', () => {
    const badgeTrophiesSelected = [
      {
        id: '1',
        icon: '/path/to/some/icon',
        isDefault: false,
        description: 'Some description',
      },
      {
        id: '2',
        icon: '/path/to/empty/icon',
        isDefault: true,
        description: 'Empty',
      },
      {
        id: '3',
        icon: '/path/to/third/icon',
        isDefault: false,
        description: 'Third description',
      },
    ]

    const badgeTrophiesUnused = [
      {
        id: '4',
        icon: '/path/to/fourth/icon',
        description: 'Fourth description',
      },
      {
        id: '5',
        icon: '/path/to/fifth/icon',
        description: 'Fifth description',
      },
    ]

    let wrapper

    beforeEach(() => {
      mocks.$store = {
        getters: {
          'auth/isModerator': () => false,
          'auth/user': {
            id: 'u23',
            badgeVerification: {
              id: 'bv1',
              icon: '/verification/icon',
              description: 'Verification description',
              isDefault: false,
            },
            badgeTrophiesSelected,
            badgeTrophiesUnused,
          },
        },
      }
      wrapper = Wrapper()
    })

    it('renders', () => {
      expect(wrapper.container).toMatchSnapshot()
    })

    describe('selecting a used badge', () => {
      beforeEach(async () => {
        const badge = screen.getByTitle(badgeTrophiesSelected[0].description)
        await fireEvent.click(badge)
      })

      it('shows remove badge button', () => {
        expect(screen.getByText('settings.badges.remove')).toBeInTheDocument()
      })

      describe('clicking remove badge button', () => {
        const clickButton = async () => {
          const removeButton = screen.getByText('settings.badges.remove')
          await fireEvent.click(removeButton)
        }

        describe('with successful server request', () => {
          beforeEach(() => {
            apolloMutateMock.mockResolvedValue({
              data: {
                setTrophyBadgeSelected: {
                  id: 'u23',
                  badgeTrophiesSelected: [
                    {
                      id: '2',
                      icon: '/path/to/empty/icon',
                      isDefault: true,
                      description: 'Empty',
                    },
                    {
                      id: '2',
                      icon: '/path/to/empty/icon',
                      isDefault: true,
                      description: 'Empty',
                    },
                    {
                      id: '3',
                      icon: '/path/to/third/icon',
                      isDefault: false,
                      description: 'Third description',
                    },
                  ],
                },
              },
            })
            clickButton()
          })

          it('calls the server', () => {
            expect(apolloMutateMock).toHaveBeenCalledWith({
              mutation: expect.anything(),
              update: expect.anything(),
              variables: {
                badgeId: null,
                slot: 0,
              },
            })
          })

          /* To test this, we would need a better apollo mock */
          it.skip('removes the badge', async () => {
            expect(wrapper.container).toMatchSnapshot()
          })

          it('shows a success message', () => {
            expect(mocks.$toast.success).toHaveBeenCalledWith('settings.badges.success-update')
          })
        })

        describe('with failed server request', () => {
          beforeEach(() => {
            apolloMutateMock.mockRejectedValue({ message: 'Ouch!' })
            clickButton()
          })

          it('shows an error message', () => {
            expect(mocks.$toast.error).toHaveBeenCalledWith('settings.badges.error-update')
          })
        })
      })
    })

    describe('no more badges available', () => {
      beforeEach(async () => {
        mocks.$store.getters['auth/user'].badgeTrophiesUnused = []
      })

      describe('selecting an empty slot', () => {
        beforeEach(async () => {
          const emptySlot = screen.getAllByTitle('Empty')[0]
          await fireEvent.click(emptySlot)
        })

        it('shows no more badges available message', () => {
          expect(wrapper.container).toMatchSnapshot()
        })
      })
    })

    describe('more badges available', () => {
      describe('selecting an empty slot', () => {
        beforeEach(async () => {
          const emptySlot = screen.getAllByTitle('Empty')[0]
          await fireEvent.click(emptySlot)
        })

        it('shows list with available badges', () => {
          expect(wrapper.container).toMatchSnapshot()
        })

        describe('clicking on an available badge', () => {
          const clickBadge = async () => {
            const badge = screen.getByText(badgeTrophiesUnused[0].description)
            await fireEvent.click(badge)
          }

          describe('with successful server request', () => {
            beforeEach(() => {
              apolloMutateMock.mockResolvedValue({
                data: {
                  setTrophyBadgeSelected: {
                    id: 'u23',
                    badgeTrophiesSelected: [
                      {
                        id: '4',
                        icon: '/path/to/fourth/icon',
                        description: 'Fourth description',
                        isDefault: false,
                      },
                      {
                        id: '2',
                        icon: '/path/to/empty/icon',
                        isDefault: true,
                        description: 'Empty',
                      },
                      {
                        id: '3',
                        icon: '/path/to/third/icon',
                        isDefault: false,
                        description: 'Third description',
                      },
                    ],
                  },
                },
              })
              clickBadge()
            })

            it('calls the server', () => {
              expect(apolloMutateMock).toHaveBeenCalledWith({
                mutation: expect.anything(),
                update: expect.anything(),
                variables: {
                  badgeId: '4',
                  slot: 1,
                },
              })
            })

            /* To test this, we would need a better apollo mock */
            it.skip('adds the badge', async () => {
              expect(wrapper.container).toMatchSnapshot()
            })

            it('shows a success message', () => {
              expect(mocks.$toast.success).toHaveBeenCalledWith('settings.badges.success-update')
            })
          })

          describe('with failed server request', () => {
            beforeEach(() => {
              apolloMutateMock.mockRejectedValue({ message: 'Ouch!' })
              clickBadge()
            })

            it('shows an error message', () => {
              expect(mocks.$toast.error).toHaveBeenCalledWith('settings.badges.error-update')
            })
          })
        })
      })
    })
  })
})
