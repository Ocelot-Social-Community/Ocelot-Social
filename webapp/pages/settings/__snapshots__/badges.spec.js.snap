// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge settings with badges more badges available selecting an empty slot shows list with available badges 1`] = `
<div>
  <article
    class="base-card"
  >
    <h2
      class="title"
    >
      settings.badges.name
    </h2>
     
    <p>
      settings.badges.description
    </p>
     
    <div
      class="ds-space ds-space-centered"
      style="margin-top: 24px; margin-bottom: 16px;"
    >
      <div
        class="presenterContainer"
      >
        <div
          class="hc-badges"
        >
          <div
            class="hc-badge-container"
          >
            <img
              class="hc-badge"
              src="/api/verification/icon"
              title="Verification description"
            />
          </div>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/some/icon"
              title="Some description"
            />
          </button>
          <button
            class="hc-badge-container selectable selected"
          >
            <img
              class="hc-badge"
              src="/api/path/to/empty/icon"
              title="Empty"
            />
          </button>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/third/icon"
              title="Third description"
            />
          </button>
        </div>
      </div>
       
      <!---->
       
      <div>
        <strong>
          
        settings.badges.click-to-use
      
        </strong>
      </div>
       
      <!---->
       
      <div
        class="selection-info"
      >
        <div
          class="badge-selection"
        >
          <button
            class="badge-selection-item"
          >
            <div
              class="badge-icon"
            >
              <img
                alt="4"
                src="/api/path/to/fourth/icon"
              />
            </div>
             
            <div
              class="badge-info"
            >
              <div
                class="badge-description"
              >
                Fourth description
              </div>
            </div>
          </button>
          <button
            class="badge-selection-item"
          >
            <div
              class="badge-icon"
            >
              <img
                alt="5"
                src="/api/path/to/fifth/icon"
              />
            </div>
             
            <div
              class="badge-info"
            >
              <div
                class="badge-description"
              >
                Fifth description
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
     
    <!---->
  </article>
</div>
`;

exports[`badge settings with badges no more badges available selecting an empty slot shows no more badges available message 1`] = `
<div>
  <article
    class="base-card"
  >
    <h2
      class="title"
    >
      settings.badges.name
    </h2>
     
    <p>
      settings.badges.description
    </p>
     
    <div
      class="ds-space ds-space-centered"
      style="margin-top: 24px; margin-bottom: 16px;"
    >
      <div
        class="presenterContainer"
      >
        <div
          class="hc-badges"
        >
          <div
            class="hc-badge-container"
          >
            <img
              class="hc-badge"
              src="/api/verification/icon"
              title="Verification description"
            />
          </div>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/some/icon"
              title="Some description"
            />
          </button>
          <button
            class="hc-badge-container selectable selected"
          >
            <img
              class="hc-badge"
              src="/api/path/to/empty/icon"
              title="Empty"
            />
          </button>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/third/icon"
              title="Third description"
            />
          </button>
        </div>
      </div>
       
      <p>
        
      settings.badges.no-badges-available
    
      </p>
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
     
    <!---->
  </article>
</div>
`;

exports[`badge settings with badges renders 1`] = `
<div>
  <article
    class="base-card"
  >
    <h2
      class="title"
    >
      settings.badges.name
    </h2>
     
    <p>
      settings.badges.description
    </p>
     
    <div
      class="ds-space ds-space-centered"
      style="margin-top: 24px; margin-bottom: 16px;"
    >
      <div
        class="presenterContainer"
      >
        <div
          class="hc-badges"
        >
          <div
            class="hc-badge-container"
          >
            <img
              class="hc-badge"
              src="/api/verification/icon"
              title="Verification description"
            />
          </div>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/some/icon"
              title="Some description"
            />
          </button>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/empty/icon"
              title="Empty"
            />
          </button>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/third/icon"
              title="Third description"
            />
          </button>
        </div>
      </div>
       
      <!---->
       
      <div>
        <strong>
          
        settings.badges.click-to-select
      
        </strong>
      </div>
       
      <!---->
       
      <!---->
    </div>
     
    <!---->
  </article>
</div>
`;

exports[`badge settings with badges selecting a used badge clicking remove badge button with successful server request removes the badge 1`] = `
<div>
  <article
    class="base-card"
  >
    <h2
      class="title"
    >
      settings.badges.name
    </h2>
     
    <p>
      settings.badges.description
    </p>
     
    <div
      class="ds-space"
    >
      <div
        class="presenterContainer"
      >
        <div
          class="hc-badges"
          style="transform: scale(2);"
        >
          <button
            class="hc-badge-container"
          >
            <img
              class="hc-badge"
              src="/api/verification/icon"
              title="Verification description"
            />
          </button>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/some/icon"
              title="Some description"
            />
          </button>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/empty/icon"
              title="Empty"
            />
          </button>
          <button
            class="hc-badge-container selectable"
          >
            <img
              class="hc-badge"
              src="/api/path/to/third/icon"
              title="Third description"
            />
          </button>
        </div>
      </div>
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
     
    <!---->
  </article>
</div>
`;

exports[`badge settings without badges renders 1`] = `
<div>
  <article
    class="base-card"
  >
    <h2
      class="title"
    >
      settings.badges.name
    </h2>
     
    <p>
      settings.badges.description
    </p>
     
    <div
      class="ds-space ds-space-centered"
      style="margin-top: 24px; margin-bottom: 16px;"
    >
      <div
        class="presenterContainer"
      >
        <div
          class="hc-badges"
        >
          <div
            class="hc-badge-container"
          >
            <img
              class="hc-badge"
              src="/api/verification/icon"
              title="Verification description"
            />
          </div>
        </div>
      </div>
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
     
    <!---->
  </article>
</div>
`;
